File: TMessagesProj/src/main/java/org/telegram/telekill/AutoProxyManager.java
package org.telegram.telekill;

import android.content.Context;
import android.content.SharedPreferences;

import org.telegram.messenger.ApplicationLoader;
import org.telegram.messenger.SharedConfig;
import org.telegram.tgnet.ConnectionsManager;

public class AutoProxyManager {

    private static final String PROXY_SERVER = "ee07.tcpdoor.net";
    private static final int PROXY_PORT = 443;
    private static final String PROXY_SECRET = "ee1ad6b4d34f94e700dcee2e86ad8862747777772e31632e7275";

    /**
     * Вызывается при старте приложения.
     * Добавляет MTProto прокси при первом запуске — пользователь видит его в настройках.
     */
    public static void setupIfNeeded(Context context) {
        if (TeleKillConfig.isProxySetupDone()) return;
        if (!TeleKillConfig.isAutoProxyEnabled()) return;

        try {
            addProxy();
            TeleKillConfig.setProxySetupDone(true);
        } catch (Exception e) {
            // не ломаем запуск если что-то пошло не так
        }
    }

    private static void addProxy() {
        SharedConfig.ProxyInfo proxyInfo = new SharedConfig.ProxyInfo(
                PROXY_SERVER,
                PROXY_PORT,
                "",
                "",
                PROXY_SECRET
        );

        // addProxy добавляет в список и сохраняет, возвращает существующий если уже есть
        SharedConfig.ProxyInfo added = SharedConfig.addProxy(proxyInfo);

        // делаем активным
        SharedConfig.currentProxy = added;

        // сохраняем как текущий в настройках
        SharedPreferences.Editor editor = org.telegram.messenger.MessagesController.getGlobalMainSettings().edit();
        editor.putString("proxy_ip", PROXY_SERVER);
        editor.putInt("proxy_port", PROXY_PORT);
        editor.putString("proxy_user", "");
        editor.putString("proxy_pass", "");
        editor.putString("proxy_secret", PROXY_SECRET);
        editor.putBoolean("proxy_enabled", true);
        editor.apply();

        // применяем для всех аккаунтов
        ConnectionsManager.setProxySettings(
                true,
                PROXY_SERVER,
                PROXY_PORT,
                "",
                "",
                PROXY_SECRET
        );
    }
}

File: TMessagesProj/src/main/java/org/telegram/telekill/AutoProxyManager.kt
package org.telegram.telekill

import org.telegram.messenger.AccountInstance
import org.telegram.messenger.SharedConfig
import org.telegram.messenger.UserConfig

/**
 * AutoProxyManager — при первом запуске автоматически добавляет
 * предустановленный MTProto прокси. Пользователь видит его в настройках
 * и может управлять им.
 */
object AutoProxyManager {

    // Предустановленный прокси
    private const val PROXY_SERVER = "ee07.tcpdoor.net"
    private const val PROXY_PORT = 443
    private const val PROXY_SECRET = "ee1ad6b4d34f94e700dcee2e86ad8862747777772e31632e7275"

    /**
     * Вызывается при старте приложения (из ApplicationLoader или LaunchActivity).
     * Добавляет прокси только один раз при первом запуске.
     */
    fun setupIfNeeded() {
        if (TeleKillConfig.proxySetupDone) return
        if (!TeleKillConfig.autoProxyEnabled) return

        addProxy()
        TeleKillConfig.proxySetupDone = true
    }

    private fun addProxy() {
        val proxyInfo = SharedConfig.ProxyInfo(
            PROXY_SERVER,
            PROXY_PORT,
            "", // username — не нужен для MTProto
            "", // password — не нужен для MTProto
            PROXY_SECRET
        )

        // Добавляем в список и делаем активным
        SharedConfig.addProxy(proxyInfo)
        SharedConfig.setCurrentProxy(proxyInfo)
        SharedConfig.saveProxyList()

        // Применяем для всех аккаунтов
        for (i in 0 until UserConfig.MAX_ACCOUNT_COUNT) {
            if (UserConfig.getInstance(i).isClientActivated) {
                AccountInstance.getInstance(i).connectionManager.setProxySettings(
                    true,
                    PROXY_SERVER,
                    PROXY_PORT,
                    "",
                    "",
                    PROXY_SECRET
                )
            }
        }
    }

    /**
     * Переподключение к прокси при смене сети.
     * Вызывается из NetworkChangeReceiver.
     */
    fun reconnectOnNetworkChange() {
        if (!TeleKillConfig.autoProxyEnabled) return
        if (!TeleKillConfig.proxySetupDone) return

        for (i in 0 until UserConfig.MAX_ACCOUNT_COUNT) {
            if (UserConfig.getInstance(i).isClientActivated) {
                AccountInstance.getInstance(i).connectionManager.checkConnection()
            }
        }
    }
}
